pkg/pgmodel/metrics/database/database.go

```diff
@@ -0,0 +1,112 @@
+package database
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/pgxconn"
+)
+
+type Engine interface {
+	Start()
+	Stop()
+}
+
+type metricsEngineImpl struct {
+	conn    pgxconn.PgxConn
+	metrics []metricQueryWrap
+	stop    chan struct{}
+}
+
+// NewEngine creates an engine that performs database metrics evaluation every evalInterval.
+// Make sure to call this only when the database is TimescaleDB. Plain Postgres
+// will cause evaluation errors.
+func NewEngine(conn pgxconn.PgxConn) *metricsEngineImpl {
+	return &metricsEngineImpl{
+		conn:    conn,
+		metrics: metrics,
+		stop:    make(chan struct{}),
+	}
+}
+
+func (e *metricsEngineImpl) register() {
+	prometheus.MustRegister(getMetrics(e.metrics)...)
+}
+
+func (e *metricsEngineImpl) unregister() {
+	prometheus.MustRegister(getMetrics(e.metrics)...)
+}
+
+func getMetrics(m []metricQueryWrap) []prometheus.Collector {
+	var metrics []prometheus.Collector
+	for i := range m {
+		metrics = append(metrics, m[i].metric)
+	}
+	return metrics
+}
+
+const sqlMetricsEvalInterval = time.Second * 3
+
+func (e *metricsEngineImpl) Start() {
+	e.register()
+	go func() {
+		evaluate := time.NewTicker(sqlMetricsEvalInterval)
+		defer evaluate.Stop()
+		numQueries := len(e.metrics)
+
+		evalContext, stopEvaluation := context.WithCancel(context.Background())
```

Comment:
This should probably have a timeout set - I guess less than `sqlMetricsEvalInterval `. Otherwise any SQL that we run below can block and we go out of eval interval